plasmapy/atomic/tests/test_classes.py

```diff
@@ -1,8 +1,143 @@
+    _neutrinos,
+    _antineutrinos,
+    _special_particles,
+)
+
 from ..classes import Particle
 
 
-def test_Particle():
-    r"""Original, temporary tests of Particle class."""
-    p = Particle('H', mass_numb=1, Z=1)
-    assert p._generation is None
+# (arg, kwargs, results_dict
+test_Particle_table = [
+
+    ('p+', {},
+     {'particle': 'p+',
+      'element': 'H',
+      'isotope': 'H-1',
+      'ion': 'p+',
+      'm': m_p,
+      'Z': 1,
+      'spin': 1/2,
+      'half_life': np.inf * u.s,
+      'atomic_number': 1,
+      'mass_number': 1,
+      'lepton_number': 0,
+      'baryon_number': 1,
+      }),
+
+    ('e-', {},
+        {'particle': 'e-',
+         'element': InvalidElementError,
+        })
+]
+
+
+@pytest.mark.parametrize("arg, kwargs, expected_dict", test_Particle_table)
+def test_Particle_class(arg, kwargs, expected_dict):
+    r"""Test required properties of items in _Particles dictionary."""
+
+    # To allow tests of a Particle class to continue after the first
+    # exception, error messages will be appended as new lines to errmsg.
+    errmsg = ""
+
+    cls = f"Particle('{arg}')"
+
+    try:
+        particle = Particle(arg)
+    except Exception as exc:
+        raise Exception(f"Unable to create {cls}.") from exc
+
+    for key in expected_dict.keys():
+        expected = expected_dict[key]
+
+        if inspect.isclass(expected) and issubclass(expected, Exception):
+            # Exceptions are expected to be raised when accessing certain
+            # attributes for some particles.  For example, accessing a
+            # neutrino's mass should raise a MissingAtomicDataError.
+            # If expected_dict[key] is an exception, then check to make
+            # sure that this exception is raised.
+            try:
+                with pytest.raises(expected):
+                    exec(f"particle.{key}")
+            except pytest.fail.Exception as exc_failed_fail:
+                errmsg += f"\n{cls}.{key} does not raise {expected}."
+            except Exception as exc_bad:
+                errmsg += (f"\n{cls}.{key} does not raise {expected} and " \
+                           f"instead raises a different exception.")
+        else:
+            try:
+                result = eval(f"particle.{key}")
+                assert result == expected
+            except AssertionError as exc_assert:
+                errmsg += f"\n{cls}.{key} does not equal {expected}."
+            except Exception as exc_general:
+                errmsg += f"\n{cls}.{key} raises an unexpected exception."
+
+    if len(errmsg) > 0:
+        raise Exception("The following problems were found for "
+                        f"Particle('{key}'):" + errmsg)
+
+
+@pytest.mark.parametrize("symbol", _special_particles)
+def test_Particle_special(symbol):
+    r"""Test the properties of special particles that do not
+    correspond to elements."""
+
+    particle = Particle(symbol)
+
+    assert particle._atomic_symbol is None, \
```

Comment:
I have a sick and twisted idea on how to stick this into `parametrize` using `lambda`s.

But perhaps that's overkill. I could try implementing it on my copy of this branch if you'd like, it shouldn't take long.

Reply:
I'm planning on deleting this test since `test_Particle_class` should do a much better job of testing this.